leveldb/session_compaction.go

```diff
@@ -62,10 +63,21 @@ func (s *session) pickCompaction() *compaction {
 		sourceLevel = v.cLevel
 		cptr := s.getCompPtr(sourceLevel)
 		tables := v.levels[sourceLevel]
-		for _, t := range tables {
-			if cptr == nil || s.icmp.Compare(t.imax, cptr) > 0 {
-				t0 = append(t0, t)
-				break
+		if cptr != nil {
+			if sourceLevel == 0 {
```

Comment:
I think we can even be more aggressive here. 

Seems level0 files are overlapped anyway, so for level0 seed file we can just pick the first one.

Reply:
ha, good idea

Reply:
so I just added a new commit according to your suggestion